File: SmartHome_WebApp/SmartHome.Model/MasterUnit.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace SmartHome.Model
{
    public class MasterUnit
    {
        public Guid Id { get; set; }

        public string CustomName { get; set; }

        public List<Design> Designs { get; set; }

        public bool IsOn { get; set; }

        [Required]
        public User Owner { get; set; }

        public List<DataSample> DataSamples { get; set; }
    }
}

File: SmartHome_WebApp/SmartHome.WebApp/ClientApp/app/components/MasterUnit.ts
export class MasterUnit {
    id: string = "00000000-0000-0000-0000-000000000000";
    eTag: string = "00000000-0000-0000-0000-000000000000";
    customName: string = "";
    isOn: boolean = false;
    ownerId: string = "00000000-0000-0000-0000-000000000000"
    constructor(id: string = "00000000-0000-0000-0000-000000000000",
        etag: string = "00000000-0000-0000-0000-000000000000",
        customName: string = "",
        isOn: boolean = false,
        ownerId: string = "00000000-0000-0000-0000-000000000000")
    {
        this.id = id;
        this.eTag = etag;
        this.customName = customName;
        this.isOn = isOn;
        this.ownerId = ownerId;
    }
}
